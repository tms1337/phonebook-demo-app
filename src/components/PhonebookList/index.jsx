import React from "react";
import PhonebookItem from "../PhonebookItem";

const PhonebookList = ({ phonebook, filter = "" }) => {
  const filteredPhoneBook = phonebook.filter(item => {
    return (
      item.name.includes(filter) ||
      item.address.includes(filter) ||
      item.phone.includes(filter)
    );
  });

  const phonebookItems = filteredPhoneBook.map(item => (
    <PhonebookItem key={item.phone} item={item} />
  ));

  return <div>{phonebookItems}</div>;
};

export default PhonebookList;
