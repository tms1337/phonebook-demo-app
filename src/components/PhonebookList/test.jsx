import React from "react";
import renderer from "react-test-renderer";

import PhonebookList from ".";

describe("PhonebookList", () => {
  it("Snapshot renders", () => {
    const items = [
      {
        name: "Heso",
        surname: "Meso",
        phone: "+007862345",
        address: "Dzembe Lembe 3"
      }
    ];

    const component = renderer.create(<PhonebookList phonebook={items} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
