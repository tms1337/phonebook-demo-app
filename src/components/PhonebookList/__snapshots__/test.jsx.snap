// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhonebookList Snapshot renders 1`] = `<div />`;
